### PR TITLE
fix license header test

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Fixes our license header check

## How this works

uses https://github.com/palantir/go-license

## How this was tested

manually + CI

### CODE REVIEW

File: ipcs/socket/socket_unix.go [hunk 1]
```diff
@@ -1,9 +1,9 @@
-//go:build !windows && !plan9 && !js
-// +build !windows,!plan9,!js
-
 // Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
 // See the file LICENSE for licensing terms.
 
+//go:build !windows && !plan9 && !js
+// +build !windows,!plan9,!js
+
```
Comment (dhrubabasu) [inline ipcs/socket/socket_unix.go:6]: https://pkg.go.dev/cmd/go#hdr-Build_constraints

```
Constraints may appear in any kind of source file (not just Go), but they must appear near the top of the file, preceded only by blank lines and other line comments. These rules mean that in Go files a build constraint must appear before the package clause.

To distinguish build constraints from package documentation, a build constraint should be followed by a blank line.
```

File: scripts/lint.sh [hunk 2]
```diff
@@ -36,17 +36,16 @@ function find_go_files {
 }
```
Comment (StephenButtolph) [inline scripts/lint.sh:36]: Should we be removing this function? (or using it?)

File: scripts/lint.sh [hunk 3]
```diff
@@ -36,17 +36,16 @@ function find_go_files {
...
 function test_license_header {
-  go install -v github.com/google/addlicense@latest
-  local target="${1}"
+  go install -v github.com/palantir/go-license@v1.25.0
   local files=()
-  while IFS= read -r line; do files+=("$line"); done < <(find_go_files "${target}")
+  while IFS= read -r line; do files+=("$line"); done < <(find . -type f -name '*.go' ! -name '*.pb.go' ! -name 'mock_*.go')
 
-  addlicense \
-  -f ./LICENSE.header \
```
Comment (StephenButtolph) [inline scripts/lint.sh:49]: I think we can delete the `LICENSE.header` file right?

File: scripts/lint.sh (hunk 2 above)
Comment (dhrubabasu) [reply scripts/lint.sh]: removed

File: scripts/lint.sh (hunk 3 above)
Comment (dhrubabasu) [reply scripts/lint.sh:49]: deleted

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**